skripchenkoveta/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+ENTER_MESSAGE = 'Enter city name: '
+ERROR_MESSAGE = 'City name can not be blank!'
+EXIT_COMMAND = 'exit!'
+MONTHS_IN_YEAR = 12
+
+def rainfall_string
+  file = File.new('data.txt', 'r:UTF-8')
+  file.read
+end
+
+def town_search(town, rainfall_string)
+  rainfall_string = rainfall_string.split("\n")
+  rainfall_string.find { |value| value.include? town }
+end
+
+def extract_numbers(rainfall_string)
+  rainfall_string.scan(/\d*\.\d/).map(&:to_f)
+end
+
+def calculate_middle_rainfall(town, rainfall_string)
+  return -1.0 unless rainfall_string.include? town
+
+  rainfall_average = 0.0
+  rainfall_string = town_search(town, rainfall_string)
```

Comment:
you pass `railnfall_string` (which contains rainfall for all towns) in `calculate_middle_rainfall`
then you reassign value to `rainfall_string` to contain rainfall for only one town

???
improve naming!))